src/modules/temperature.cpp

```diff
@@ -16,12 +17,42 @@ namespace modules {
   temperature_module::temperature_module(const bar_settings& bar, string name_)
       : timer_module<temperature_module>(bar, move(name_)) {
     m_zone = m_conf.get(name(), "thermal-zone", 0);
+    m_zone_type = m_conf.get(name(), "zone-type", ""s);
     m_path = m_conf.get(name(), "hwmon-path", ""s);
     m_tempbase = m_conf.get(name(), "base-temperature", 0);
     m_tempwarn = m_conf.get(name(), "warn-temperature", 80);
     set_interval(1s);
     m_units = m_conf.get(name(), "units", m_units);
 
+    if (!m_zone_type.empty()) {
+      vector<int> zone_list;
+      vector<string> thermal_list = file_util::list_files("/sys/class/thermal/");
+      for (auto &thermal: thermal_list) {
+        string zone_str = regex_replace(thermal, std::regex("thermal_zone([0-9]+)"), string("$1"), std::regex_constants::format_no_copy);
+        if (!zone_str.empty()) {
+          zone_list.push_back(stoi(zone_str));
+        }
+      }
```

Comment:
Optional, but could make this code cleaner: Instead of extracting the zone number, we could `file_util::glob` for `/sys/class/thermal/thermal_zone*` and just add `/type` for getting the type and `/temp` to get `m_path`, the zone number isn't really needed for the module itself.

Pick whatever solution you like better.

Reply:
I kept the zone number as an effort to keep behavior consistent for when there are multiple sensors with the same name. We know we'll always pick the one with lowest number, which depending on hardware and kernel may or may not be random.

Do you still want me to change this?

Reply:
Oh that's a good point. I think sorting the paths should be sufficient to keep consistency (even if it doesn't guarantee that the lowest number is picked).

It's up to you, whichever solution you like better

Reply:
https://github.com/polybar/polybar/pull/2752/files#diff-4ecfc5580e380c6cc659596ee5bb12d8f8df8ceea72dae9559e1494503d54691R28